src/DonationForms/Actions/DispatchDonateControllerDonationCreatedListeners.php

```diff
@@ -6,15 +6,18 @@
 use Give\DonationForms\Listeners\StoreCustomFields;
 use Give\DonationForms\Listeners\TemporarilyReplaceLegacySuccessPageUri;
 use Give\Donations\Models\Donation;
+use Give\Framework\FieldsAPI\Exceptions\NameCollisionException;
+use Give\Subscriptions\Models\Subscription;
 
 class DispatchDonateControllerDonationCreatedListeners
 {
     /**
      * @since 3.0.0
+     * @throws NameCollisionException
      */
-    public function __invoke(DonateControllerData $formData, Donation $donation)
+    public function __invoke(DonateControllerData $formData, Donation $donation, Subscription $subscription = null)
```

Comment:
This can also be nullable instead of optional.

Reply:
Resolved 2438b062